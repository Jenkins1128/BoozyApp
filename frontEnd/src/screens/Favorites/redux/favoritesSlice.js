import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';
const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get(`${baseUrl}/get`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
