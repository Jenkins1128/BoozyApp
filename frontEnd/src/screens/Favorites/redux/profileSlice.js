import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/get');
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
