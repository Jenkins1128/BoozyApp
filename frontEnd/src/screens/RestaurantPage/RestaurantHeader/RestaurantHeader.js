import React from 'react';
import { View, Text, StyleSheet, TouchableWithoutFeedback, TouchableOpacity, Image, ImageBackground } from 'react-native';
import StarRating from 'react-native-star-rating';
import { AntDesign } from '@expo/vector-icons';

const RestaurantHeader = ({ state, onStarRatingPressed, favorite, dismiss }) => {
	return (
		<TouchableWithoutFeedback testID={'header'} onPress={dismiss}>
			<View style={styles.header}>
				<Image testID={'logoImage'} source={{ uri: state.restaurantImage }} style={styles.logoImage}></Image>
				<View testID={'titleHeader'} style={styles.titleHeader}>
					<Text testID={'restaurantName'} style={styles.restaurantName}>
						{state.restaurantName}
					</Text>
					<TouchableOpacity testID={'favoriteButton'} onPress={favorite} style={styles.favoriteButton}>
						<AntDesign name='heart' style={{ color: state.favoriteColor, fontSize: 30 }} />
					</TouchableOpacity>
				</View>
				<View testID={'titleBody'} style={styles.titleBody}>
					<View testID={'infoContainer'} style={styles.infoContainer}>
						<Text testID={'categoriesText'} style={styles.categoriesText}>
							{state.allCategories} • {state.rating}★
						</Text>
						<Text testID={'phoneNumberText'} style={styles.phoneNumberText}>
							{state.phoneNumber} • {state.reviews} Yelp Reviews
						</Text>
					</View>
					<View testID={'starRatingContainer'} style={styles.starRatingContainer}>
						<StarRating
							disabled={false}
							emptyStar={'ios-star-outline'}
							fullStar={'ios-star'}
							halfStar={'ios-star-half'}
							iconSet={'Ionicons'}
							maxStars={5}
							disabled={state.rated}
							rating={state.starCount}
							selectedStar={(rating) => onStarRatingPressed(rating)}
							fullStarColor={'yellow'}
							ratingBackgroundColor='white'
							starSize={30}
						/>
					</View>
				</View>
			</View>
		</TouchableWithoutFeedback>
	);
};

export default RestaurantHeader;

const styles = StyleSheet.create({
	header: {
		flex: 1,
		width: '100%',
		justifyContent: 'center',
		backgroundColor: '#EB8873'
	},
	titleHeader: {
		flexDirection: 'row',
		justifyContent: 'space-between',
		alignItems: 'center',
		marginBottom: 10
	},
	logoImage: {
		flex: 2,
		width: '100%',
		height: '100%'
	},
	restaurantName: {
		color: 'white',
		fontSize: 25,
		fontWeight: 'bold',
		marginLeft: 10
	},
	categoriesText: {
		fontSize: 14,
		marginBottom: 0,
		color: 'white'
	},
	phoneNumberText: {
		fontSize: 14,
		marginBottom: 0,
		color: 'white'
	},
	favoriteButton: {
		marginBottom: 0,
		margin: 10
	},
	titleBody: {
		flex: 1,
		marginHorizontal: 15
	},
	infoContainer: {
		flex: 1,
		marginBottom: 0
	},
	starRatingContainer: {
		flex: 1,
		paddingBottom: 0
	}
});
