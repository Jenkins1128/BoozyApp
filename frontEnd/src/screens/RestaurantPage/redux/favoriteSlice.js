import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {}
};

export const favoriteAsync = createAsyncThunk('favoriteAsync/status', async (data, { rejectWithValue }) => {
	const { restaurantId, name } = data;
	try {
		//favorite restaurant
		await axios.post(`${baseUrl}/user/add`, {
			restaurantId: restaurantId,
			name: name
		});
		//add restaurants to your favorites
		const response = await axios.post(`${baseUrl}/favorites`, {
			restaurantId: restaurantId,
			name: name
		});
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

const favoriteSlice = createSlice({
	name: 'favoriteAsync',
	initialState,
	reducers: {
		resetFavorite: (state) => {
			state.currentState.favoriteRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(favoriteAsync.fulfilled, (state, { payload }) => {
				Object.assign(state.currentState, {
					favoriteColor: payload[0]['contains'] ? 'red' : 'white',
					favoriteRequestStatus: 'fulfilled'
				});
			})
			.addCase(favoriteAsync.rejected, (state) => {
				state.currentState.favoriteRequestStatus = 'rejected';
			});
	}
});

export const { resetFavorite } = favoriteSlice.actions;

export const selectFavoriteState = (state) => state.favorite.currentState;
export default favoriteSlice.reducer;
