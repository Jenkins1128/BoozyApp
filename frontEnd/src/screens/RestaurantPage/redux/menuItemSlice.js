import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {}
};

export const postMenuItemAsync = createAsyncThunk('postMenuItemAsync/status', async (data, { rejectWithValue }) => {
	const { description, price, restaurantName, restaurantId } = data;
	try {
		const response = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/user/' + restaurantId + '/menu', {
			content: description,
			price: price,
			name: restaurantName
		});
		return response.status;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

const postMenuItemSlice = createSlice({
	name: 'postMenuItemAsync',
	initialState,
	reducers: {
		updatePrice: (state, { payload }) => {
			state.currentState = { ...state.currentState, price: payload.price };
		},
		updateDescription: (state, { payload }) => {
			state.currentState = { ...state.currentState, description: payload.description };
		},
		resetMenuItem: (state) => {
			state.currentState = { ...state.currentState, price: '', description: '', postMenuItemRequestStatus: 'idle' };
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(postMenuItemAsync.fulfilled, (state) => {
				state.currentState = { ...state.currentState, postMenuItemRequestStatus: 'fulfilled' };
			})
			.addCase(postMenuItemAsync.rejected, (state) => {
				state.currentState = { ...state.currentState, postMenuItemRequestStatus: 'rejected' };
			});
	}
});

export const { updatePrice, updateDescription, resetMenuItem } = postMenuItemSlice.actions;

export const selectMenuItemState = (state) => state.menuItem.currentState;
export default postMenuItemSlice.reducer;
