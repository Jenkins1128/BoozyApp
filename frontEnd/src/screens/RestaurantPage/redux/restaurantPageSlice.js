import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl, defaultRestaurantImageUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {
		restaurantId: '',
		restaurantName: '',
		restaurantImage: defaultRestaurantImageUrl,
		phoneNumber: '',
		allCategories: '',
		rating: 0.0,
		reviews: 0,
		favoriteColor: 'white',
		changeFavoritedColor: '',
		starCount: 0,
		favorited: false,
		rated: false,
		price: 0.0,
		description: '',
		menuItemArray: []
	}
};

export const getMenuItemsAsync = createAsyncThunk('getMenuItemsAsync/status', async (restaurantId, { rejectWithValue }) => {
	try {
		const response = await axios.post(`${baseUrl}/menu/list`, {
			restaurantId: restaurantId
		});
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const restaurantPageSlice = createSlice({
	name: 'getMenuItemsAsync',
	initialState,
	reducers: {
		updateState: (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantId: payload.params.id,
				restaurantName: payload.params.name,
				restaurantImage: payload.params.img,
				phoneNumber: payload.params.phone,
				allCategories: payload.params.allCategories,
				rating: payload.params.rating,
				reviews: payload.params.reviews,
				rated: payload.params.alreadyRated,
				favorited: payload.params.alreadyFavorited,
				starCount: 0
			});
		},
		updateFavoriteColor: (state, { payload }) => {
			state.currentState.favoriteColor = payload.color;
		},
		updateStarCount: (state, { payload }) => {
			state.currentState.starCount = payload.starCount;
		},
		updateMenuItems: (state, { payload }) => {
			state.currentState.menuItemArray.push(payload.menuItem);
		},
		resetRestaurantPageRequestStatus: (state) => {
			state.currentState.restaurantPageRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(getMenuItemsAsync.fulfilled, (state, { payload }) => {
				Object.assign(state.currentState, {
					restaurantPageRequestStatus: 'fulfilled',
					menuItemArray: payload
				});
			})
			.addCase(getMenuItemsAsync.rejected, (state) => {
				state.currentState.restaurantPageRequestStatus = 'rejected';
			});
	}
});

export const { updateState, updateFavoriteColor, updateStarCount, updateMenuItems, resetRestaurantPageRequestStatus } = restaurantPageSlice.actions;

export const selectRestaurantPageState = (state) => state.restaurantPage.currentState;
export default restaurantPageSlice.reducer;
