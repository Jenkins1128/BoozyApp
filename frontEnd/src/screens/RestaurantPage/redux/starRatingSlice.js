import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {}
};

export const starRatingAsync = createAsyncThunk('starRatingAsync/status', async (data, { rejectWithValue }) => {
	const { rating, restaurantId, restaurantName } = data;
	try {
		const response = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/user/' + restaurantId + '/restaurants', {
			content: rating,
			name: restaurantName
		});
		return response.status;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const starRatingSlice = createSlice({
	name: 'starRatingAsync',
	initialState,
	reducers: {
		resetStarRatingRequestStatus: (state) => {
			state.currentState.starRatingRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(starRatingAsync.fulfilled, (state) => {
				state.currentState.starRatingRequestStatus = 'fulfilled';
			})
			.addCase(starRatingAsync.rejected, (state) => {
				state.currentState.starRatingRequestStatus = 'rejected';
			});
	}
});

export const { resetStarRatingRequestStatus } = starRatingSlice.actions;

export const selectStarRatingState = (state) => state.starRating.currentState;
export default starRatingSlice.reducer;
