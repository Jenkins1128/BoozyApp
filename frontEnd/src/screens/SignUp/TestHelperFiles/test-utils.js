import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const INITIAL_STATE = {
	signup: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	},
	login: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	}
}; // Or use value from root reducer
const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const reduxRender = (ui, options) =>
	render(ui, {
		wrapper: ({ children }) => <Provider store={store}>{children}</Provider>,
		...options
	});

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { reduxRender, store };
