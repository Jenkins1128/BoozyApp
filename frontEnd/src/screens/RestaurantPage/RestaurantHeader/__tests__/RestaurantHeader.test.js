import React from 'react';
import { render, cleanup, fireEvent } from '../../../../TestHelperFiles/redux/test-utils';
import RestaurantHeader from '../RestaurantHeader';

describe('<RestaurantHeader />', () => {
	let state;
	let onStarRatingPressed, favorite, dismiss;
	let rendered;

	beforeEach(() => {
		state = {
			restaurantImage: '',
			restaurantName: '',
			favoriteColor: '',
			allCategories: '',
			rating: 4,
			phoneNumber: '',
			reviews: '',
			rated: false,
			starCount: 0
		};
		(onStarRatingPressed = jest.fn()), (favorite = jest.fn()), (dismiss = jest.fn());
		rendered = render(<RestaurantHeader state={state} onStarRatingPressed={onStarRatingPressed} favorite={favorite} dismiss={dismiss} />);
	});

	afterEach(cleanup);

	//snapshot
	it('should render correctly', () => {
		expect(rendered).toMatchSnapshot();
	});

	//content
	it('should wrap flexible wrapper over view and list', () => {
		const headerComponent = rendered.getByTestId('header');
		const given = headerComponent.props.style;
		const result = {
			flex: 1,
			width: '100%',
			justifyContent: 'center',
			backgroundColor: '#EB8873'
		};
		expect(result).toMatchObject(given);
	});

	it('should render titleHeader styles correctly', () => {
		const titleHeaderComponent = rendered.getByTestId('titleHeader');
		const given = titleHeaderComponent.props.style;
		const result = {
			flexDirection: 'row',
			justifyContent: 'space-between',
			alignItems: 'center',
			marginBottom: 10
		};
		expect(result).toMatchObject(given);
	});

	it('should render logoImage styles correctly', () => {
		const logoImageComponent = rendered.getByTestId('logoImage');
		const given = logoImageComponent.props.style;
		const result = {
			flex: 2,
			width: '100%',
			height: '100%'
		};
		expect(result).toMatchObject(given);
	});

	it('should render restaurantName styles correctly', () => {
		const restaurantNameComponent = rendered.getByTestId('restaurantName');
		const given = restaurantNameComponent.props.style;
		const result = {
			color: 'white',
			fontSize: 25,
			fontWeight: 'bold',
			marginLeft: 10
		};
		expect(result).toMatchObject(given);
	});

	it('should render categoriesText styles correctly', () => {
		const categoriesTextComponent = rendered.getByTestId('categoriesText');
		const given = categoriesTextComponent.props.style;
		const result = {
			fontSize: 14,
			marginBottom: 0,
			color: 'white'
		};
		expect(result).toMatchObject(given);
	});

	it('should render phoneNumberText styles correctly', () => {
		const phoneNumberTextComponent = rendered.getByTestId('phoneNumberText');
		const given = phoneNumberTextComponent.props.style;
		const result = {
			fontSize: 14,
			marginBottom: 0,
			color: 'white'
		};
		expect(result).toMatchObject(given);
	});

	it('should render favoriteButton styles correctly', () => {
		const favoriteButtonComponent = rendered.getByTestId('favoriteButton');
		const given = favoriteButtonComponent.props.style;
		const result = {
			marginBottom: 0,
			margin: 10,
			opacity: 1
		};
		expect(result).toMatchObject(given);
	});

	it('should render titleBody styles correctly', () => {
		const titleBodyComponent = rendered.getByTestId('titleBody');
		const given = titleBodyComponent.props.style;
		const result = {
			flex: 1,
			marginHorizontal: 15
		};
		expect(result).toMatchObject(given);
	});

	it('should render infoContainer styles correctly', () => {
		const infoContainerComponent = rendered.getByTestId('infoContainer');
		const given = infoContainerComponent.props.style;
		const result = {
			flex: 1,
			marginBottom: 0
		};
		expect(result).toMatchObject(given);
	});

	it('should render starRatingContainer styles correctly', () => {
		const starRatingContainerComponent = rendered.getByTestId('starRatingContainer');
		const given = starRatingContainerComponent.props.style;
		const result = {
			flex: 1,
			paddingBottom: 0
		};
		expect(result).toMatchObject(given);
	});

	//interaction
	it('should fire dismiss button event on TouchableWithoutFeedback', () => {
		const dismissButton = rendered.getByTestId('header');
		fireEvent.press(dismissButton);
		expect(dismiss).toHaveBeenCalledTimes(1);
	});

	it('should fire favorite button event', () => {
		const favoriteButton = rendered.getByTestId('favoriteButton');
		fireEvent.press(favoriteButton);
		expect(favorite).toHaveBeenCalledTimes(1);
	});
});
