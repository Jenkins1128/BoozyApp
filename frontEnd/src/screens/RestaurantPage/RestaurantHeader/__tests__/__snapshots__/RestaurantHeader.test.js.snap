// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RestaurantHeader /> should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#EB8873",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
  testID="header"
>
  <Image
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Object {
        "flex": 2,
        "height": "100%",
        "width": "100%",
      }
    }
    testID="logoImage"
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
      }
    }
    testID="titleHeader"
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 25,
          "fontWeight": "bold",
          "marginLeft": 10,
        }
      }
      testID="restaurantName"
    >
      
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "margin": 10,
          "marginBottom": 0,
          "opacity": 1,
        }
      }
      testID="favoriteButton"
    >
      <Text />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 15,
      }
    }
    testID="titleBody"
  >
    <View
      style={
        Object {
          "flex": 1,
          "marginBottom": 0,
        }
      }
      testID="infoContainer"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "marginBottom": 0,
          }
        }
        testID="categoriesText"
      >
        
         • 
        4
        ★
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "marginBottom": 0,
          }
        }
        testID="phoneNumberText"
      >
        
         • 
        
         Yelp Reviews
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingBottom": 0,
        }
      }
      testID="starRatingContainer"
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          forwardedRef={[Function]}
          style={Object {}}
        >
          <View
            accessibilityRole="button"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
        </View>
        <View
          forwardedRef={[Function]}
          style={Object {}}
        >
          <View
            accessibilityRole="button"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
        </View>
        <View
          forwardedRef={[Function]}
          style={Object {}}
        >
          <View
            accessibilityRole="button"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
        </View>
        <View
          forwardedRef={[Function]}
          style={Object {}}
        >
          <View
            accessibilityRole="button"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
        </View>
        <View
          forwardedRef={[Function]}
          style={Object {}}
        >
          <View
            accessibilityRole="button"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
